src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -1784,9 +1762,6 @@ private int tryExecute(final Runnable runnable) {
                     }
                     // otherwise the consumer gave up or was exited already, so fall out and...
                 }
-                if (UPDATE_STATISTICS) spinMisses.increment();
-                // retry with new tail(snapshot) as was foretold
-                tail = this.tail;
             } else if (tailNext == null) {
```

Comment:
We could prioritize the null check as well, but I think this case is rare. There are no available threads and the queue is empty.

Reply:
If the queue is empty for real, the instanceof won't fetch the class word because the null check is performed earlier. The interesting part is that we save it to be performed twice probably (not sure)

Reply:
That's a good point, the earlier null check may produce simpler optimized code. I'll reorder this later today.